Fix slack GitHub action

zaneselvans opened the pull request:
# Overview

Update our usage of `slack-github-action` to be compatible with the new v2.0 of that action.

# Testing

The same action failed in the main PUDL repo and this fix worked there. Just updating the syntax here. Have not actually tested (but the old way would definitely fail).

e-belfer reviewed (changes requested):
Running the `early_final_release_checker` action [returned something](https://catalystcooperative.slack.com/archives/C03FHB9N0PQ/p1732112369041809) that's clearly not formatted correctly.

> {"attachments":[{"blocks":[{"type":"header","text":{"type":"plain_text","text":"Archiver Run Outcomes"}},{"type":"section","text":{"type":"mrkdwn","text":"Unchanged"}},{"type":"section","text":{"type":"mrkdwn","text":"[eia860](https://zenodo.org/deposit/14192827)\nNo changes."}},{"type":"section","text":{"type":"mrkdwn","text":"[eia861](https://zenodo.org/deposit/14192826)\nNo changes."}},{"type":"section","text":{"type":"mrkdwn","text":"[eia923](https://zenodo.org/deposit/14192828)\nNo changes."}}]}]}

zaneselvans commented:
Okay I think this is working correctly now.  I switched to using the `blocks` input to the [`chat.postMessage()` method](https://api.slack.com/methods/chat.postMessage) and modified the script to generate an array of blocks.

zaneselvans commented on scripts/make_slack_notification_message.py:
```diff
@@ -122,16 +122,10 @@ def section_block(text: str) -> dict:
-                    }
-                ]
-            },
+            [header_block("Archiver Run Outcomes")]
+            + failed_blocks
+            + changed_blocks
+            + unchanged_blocks,
```
An array of blocks, as required for the `blocks` input to the [`chat.postMessage()` method](https://api.slack.com/methods/chat.postMessage).

e-belfer reviewed (approved):
Looks like the slackbot worked as expected this time, thanks for fixing it!
